components/net/http_loader.rs

```diff
@@ -1735,6 +1742,62 @@ impl Drop for ResponseEndTimer {
     }
 }
 
+fn prompt_user_for_credentials(
+    embedder_proxy: &Mutex<Option<EmbedderProxy>>,
+) -> Option<(Option<String>, Option<String>)> {
+    let Ok(guard) = embedder_proxy.lock() else {
```

Comment:
should I just use `unwrap` on the `lock`?  I don't know if it's ok to panic here.

Reply:
Yeah, if locks are poisoned our network stack is going to fail imminently. Unwrapping is fine.